pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go

```diff
@@ -167,7 +167,7 @@ type (
 	// error over calling `t.Fatal` directly. The error is handled by
 	// the mixedversion framework and better error messages are produced
 	// as a result.
-	userFunc      func(context.Context, *logger.Logger, *rand.Rand, *Helper) error
+	UserFunc      func(context.Context, *logger.Logger, *rand.Rand, *Helper) error
```

Comment:
Do we really need to make this public in this PR? All of the current references could be removed if the methods on `cdcMixedVersionTester` implemented `userFunc` directly. That would even make them more readable, IMO.